x-pack/plugins/observability/server/lib/rules/custom_threshold/custom_threshold_executor.ts

```diff
@@ -333,8 +334,15 @@ export const createCustomThresholdExecutor = ({
           alertsLocator,
           basePath.publicBaseUrl
         ),
-        group: groupByKeysObjectForRecovered[recoveredAlertId],
+        group,
         timestamp: startedAt.toISOString(),
+        viewInAppUrl: getViewInAppUrl({
```

Comment:
We didn't have the viewInAppUrl in the alert context before? So this PR is not only about adding test,right?

Reply:
I've added the viewInApp URL in a previous [PR](https://github.com/elastic/kibana/pull/171985); I just saw adding it for the recovered alerts is also possible, so I covered it in this PR, and I also added a test for it.

Reply:
Nice. I think updating the PR title to include that, would be good.

Reply:
Done!